lib/fast_jsonapi/serialization_core.rb

```diff
@@ -72,13 +72,11 @@ def links_hash(record, params = {})
         end
       end
 
-      def attributes_hash(record, params = {})
-        attributes_to_serialize.each_with_object({}) do |(key, method), attr_hash|
-          attr_hash[key] = if method.is_a?(Proc)
-            method.arity == 1 ? method.call(record) : method.call(record, params)
-          else
-            record.public_send(method)
-          end
+      def attributes_hash(record, fieldset = nil, params = {})
+        attributes = attributes_to_serialize
+        attributes = attributes.slice(*fieldset) if fieldset.present?
+        attributes.each_with_object({}) do |(_k, attribute), hash|
+          attribute.serialize(record, params, hash)
```

Comment:
I'm losing track of how a Proc is evaluated or `call`ed?

Reply:
haha yeah this was a tricky one to track down. https://github.com/Netflix/fast_jsonapi/blob/5ff3fa97dae4b6be51bf8ec315cb427667c4ce96/lib/fast_jsonapi/attribute.rb#L21

Reply:
I didn't actually "add" any net new logic of my own, I just added more missing pieces from `v1.3` that Nate didn't port over. So instead of defining each attribute as `block || method_name`, we're  defining them as `Attribute` objects which handle the proc and block logic

Reply:
👍